### PR TITLE
Fix text selection in collection description

### LINKED ISSUE #1473: i cant select the text in collection description / or click on the link
### What happened?

i cant select the text in collection description / or click on the link

### Please reproduce in steps

/

### Expected Behavior

/

### What browsers are you seeing the problem on?

Chrome

### Are you logged in?

Yes

### At which address did you encounter bug?

nft.kodadot.xyz / kodadot.xyz

### Screenshots

/

### Relevant log output

```shell
/
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MubarakSULAYMAN**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇  _ Do a quick check before the merge. 

### PR type
- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### Before submitting Pull Request, please make sure:
- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted screenshot of demonstrated change in this PR

### Optional
- [ ] I've tested it at `/rmrk/collection/32888663bc9f50ee0e-6X3X9`
- [x] I've tested PR on mobile and everything works
- [ ] I found edge cases

### What's new?
- [x] PR closes #1473
- [x] Fix text selection in collection description

### Had issue bounty label ?
- [x] Fill up your KSM address: [Payout](https://kodadot.xyz/transfer/?target=115MLM3vGnaRLQ5LShjqcTrHM8z4PbhVMEBgXTVjsxixdd6)

### Community participation
- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot
[x] My fix has changed **something** on UI, a screenshot for others, is best to understand changes.
- Image showing text selection in collection description section
Image showing text selection in collection description section

### CODE REVIEW

File: components/rmrk/Gallery/CollectionItem.vue [hunk 1]
```diff
@@ -375,3 +375,14 @@ export default class CollectionItem extends mixins(ChainMixin, PrefixMixin) {
   color: transparent;
 }
 </style>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
+
+.description {
```
Comment (vikiival) [inline components/rmrk/Gallery/CollectionItem.vue:382]:
```suggestion
.is-selectable {
```

File: components/rmrk/Gallery/CollectionItem.vue [hunk 2]
```diff
@@ -53,7 +53,7 @@
           hidden="collapse.collection.description.hide"
           :open-on-default="!compactCollection"
         >
-          <VueMarkdown :source="description" />
+          <VueMarkdown :source="description" class="description" />
```
Comment (vikiival) [inline components/rmrk/Gallery/CollectionItem.vue:56]:
```suggestion
          <VueMarkdown :source="description" class="is-selectable" />
```

File: components/rmrk/Gallery/CollectionItem.vue [hunk 3]
```diff
@@ -375,3 +375,14 @@ export default class CollectionItem extends mixins(ChainMixin, PrefixMixin) {
   color: transparent;
 }
 </style>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
```
Comment (vikiival) [inline components/rmrk/Gallery/CollectionItem.vue:380]: you do not use any variables here

**Review by vikiival — CHANGES_REQUESTED** [review]
May Also thinking to wrap `VueMarkdown` to `SelectableMarkdown`

File: components/rmrk/Gallery/CollectionItem.vue (hunk 3 above)
Comment (MubarakSULAYMAN) [reply components/rmrk/Gallery/CollectionItem.vue]: No, I removed it already.

**Review by roiLeo — CHANGES_REQUESTED** [review]
> May Also thinking to wrap `VueMarkdown` to `SelectableMarkdown`

Why? VueMarkdown component works well, text is selectable in GalleryItem component. 
Something wrong with CollapseWrapper component, see css rule:
```css
user-select: none;
```
could be fixed by changing to:
```css
user-select: text;
```

**Comment by MubarakSULAYMAN** [comment]
@roiLeo, it seems @vikiival wants a [reusable component](https://github.com/MubarakSULAYMAN/nft-gallery/blob/patch-1/components/shared/SelectableMarkdown.vue) in case of possible future usages.

**Comment by roiLeo** [comment]
I feel this `SelectableMarkdown` wrapper is not needed as `VueMarkdown` text is already selectable by default.
IMO, fix should be on `CollapseWrapper` component.
@vikiival What do you think?

**Comment by vikiival** [comment]
Agree then. I though that vue markdown is not selectable

**Comment by MubarakSULAYMAN** [comment]
@vikiival and @roiLeo, this broken check is not in my implementation. Kindly review.

File: components/shared/collapse/CollapseWrapper.vue [hunk 4]
```diff
@@ -1,13 +1,18 @@
 <template>
   <div>
-    <b-collapse :open="isOpen" :position="position" aria-id="contentIdForA11y1" class="is-unselectable">
+    <b-collapse
+      :open="isOpen"
+      :position="position"
+      aria-id="contentIdForA11y1"
+      :class="[isSelectable ? 'is-selectable' : 'is-unselectable']"
```
Comment (roiLeo) [inline components/shared/collapse/CollapseWrapper.vue:7]:
```suggestion
      :class="{ 'is-unselectable': !isSelectable }"
```

File: components/shared/collapse/CollapseWrapper.vue [hunk 5]
```diff
@@ -32,3 +38,12 @@ export default class CollapseWrapper extends Vue {
...
 </script>
+
+<style scoped>
+.is-selectable {
+    -webkit-user-select: text;
+    -moz-user-select: text;
+    -ms-user-select: text;
+    user-select: text;
+}
+</style>
```
Comment (roiLeo) [inline components/shared/collapse/CollapseWrapper.vue:49]:
```suggestion
```
I don't think this is needed

File: components/shared/collapse/CollapseWrapper.vue [hunk 6]
```diff
@@ -24,6 +29,7 @@ export default class CollapseWrapper extends Vue {
   @Prop({ type: String, default: 'Hide' }) hidden!: string
   @Prop({ type: Boolean, default: false }) private readonly openOnDefault!: boolean
   @Prop(Boolean) bottom!: boolean
+  @Prop({ type: Boolean, default: false }) isSelectable!: Boolean
```
Comment (roiLeo) [inline components/shared/collapse/CollapseWrapper.vue:32]:
```suggestion
  @Prop({ type: Boolean, default: false }) isSelectable!: boolean
```

File: components/rmrk/Gallery/CollectionItem.vue [hunk 7]
```diff
@@ -52,6 +52,7 @@
           visible="collapse.collection.description.show"
           hidden="collapse.collection.description.hide"
           :open-on-default="!compactCollection"
+          :isSelectable="true"
```
Comment (roiLeo) [inline components/rmrk/Gallery/CollectionItem.vue:55]:
```suggestion
          isSelectable
```

**Review by roiLeo — APPROVED** [review]
lgtm

**Comment by yangwao** [comment]
@MubarakSULAYMAN let's update your PR to include KSM address

**Comment by MubarakSULAYMAN** [comment]
@yangwao, can you give more details?

**Comment by yangwao** [comment]
> @yangwao, can you give more details?

https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address

**Comment by yangwao** [comment]
Perfect, I’ve sent the payout https://kusama.subscan.io/extrinsic/0x0795276eff93f1095da6313e0ae4778e353e78c9a8a94d0016a85f34fb2c3ffa

Let’s grab another issue and get rewarded! 
https://github.com/kodadot/nft-gallery/issues/

**Pull request merged**